Bottom: 2c8fa80b524369812f2003418bddec36c6c7cf5e
Top:    77586aea0fdb458cc8d360472bbd503381a825dd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-28 15:06:38 +0100

hid/gtk: Remove dead code (due to bitrot) from pinout_zoom_fit() function

The code tried to compare against the uninitialised, unused Output.Width
and Output.Height. Assuming these were initialised to zero by the compiler,
the conditional would always cause the function to return FALSE.

Remove the conditional, and drop the boolean return value from this
function, short-cutting the caller to the utilised code-path.

This removes the (long broken) feature where by the natural size
(in pixels) of the pinout preview would be reduced if it was greater than
3/4 of the viewport size (in pixels) of the active PCB.


---

diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index 6cd8069..2a205fa 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -52,7 +52,7 @@
 RCSID ("$Id$");
 
 
-static gboolean
+static void
 pinout_zoom_fit (GhidPinoutPreview * pinout, gint zoom)
 {
   pinout->zoom = zoom;
@@ -69,11 +69,6 @@ pinout_zoom_fit (GhidPinoutPreview * pinout, gint zoom)
   pinout->h_pixels = (gint) (pinout->scale *
 			     (pinout->element.BoundingBox.Y2 -
 			      pinout->element.BoundingBox.Y1));
-
-  if (pinout->w_pixels > 3 * Output.Width / 4 ||
-      pinout->h_pixels > 3 * Output.Height / 4)
-    return FALSE;
-  return TRUE;
 }
 
 
@@ -128,8 +123,7 @@ pinout_set_data (GhidPinoutPreview * pinout, ElementType * element)
 		       Settings.PinoutOffsetY -
 		       pinout->element.BoundingBox.Y1);
 
-  if (!pinout_zoom_fit (pinout, 2))
-    pinout_zoom_fit (pinout, 3);
+  pinout_zoom_fit (pinout, 3);
 
   ELEMENTLINE_LOOP (&pinout->element);
   {
